vdev/src/commands/install_git_hooks.rs

```diff
@@ -0,0 +1,40 @@
+use anyhow::{Result, Ok};
+use std::process::Command;
+use crate::git;
+use crate::app::CommandExt as _;
+// use std::{env, path::PathBuf};
+
+/// Install a Git commit message hook that verifies
+/// that all commits have been signed off.
+#[derive(clap::Args, Debug)]
+#[command()]
+pub struct Cli {
+    /// The mode argument. Can be used to control which hook(s) are installed,
+    /// with the default being to install all available hooks.
+    mode: Option<String>
+}
+
+impl Cli {
+    pub fn exec(self) -> Result<()> {
+        let mode = self.mode.unwrap_or_else(|| "all".to_string());
+        let git_dir = git::get_git_dir()?;
+
+        // Create a new directory named hooks in the .git directory if it
+        // doesn't already exist.
+        let mut command = Command::new("mkdir");
+        command.in_repo();
+        command.args(["-p", &format!("{git_dir}/hooks")]);
+
+        command.check_run()?;
+
+        // Copy the script scripts/signoff-git-hook.sh to the
+        // .git/hooks/commit-msg file
+        if mode == "all" || mode == "signoff" {
```

Comment:
I was thinking adding this feature would be nice to have a pre-commit hook that runs cargo vdev clippy and cargo vdev fmt 

I looked into implementing the enum for all and signoff but now realize users might want differing pre-commit hooks and different commit-msg hooks. And given that github only supports one TYPE of hook per script eg one commit-msg script then it makes no sense adding the ALL field since that might confuse contributors to thinking the behaviour will be every commit-msg script in vdev will be installed when that won't be the case. The ALL field would have to have a separate script that calls the other scripts which I introduced a chicken and the egg issue since I am deleting the signoff-git-hooks and putting them as constants in the rust program, making it not possible to create a script to call other scripts to support the ALL field in case of multiple commit-msg hooks being present in the future. 

I can create a PR / feature request for that pre-commit hook I mentioned but that will be tracked separately.